From 90de629498e617ce2103bfbb4ac948c8ea295feb Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 5 Dec 2024 14:10:41 +0800
Subject: [PATCH 178/273] AOSCOS: MIPS: loongson64/smp: use {add,remove}_cpu()
 to replace cpu_{up,down}()

With commit 93ef1429e556 ("cpu/hotplug: Add new {add,remove}_cpu()
functions"), the "userspace safe" add_cpu() and remove_cpu() were
introduced to replace cpu_up() and cpu_down(), respectively. The latter
two functions were then hidden since commit 33c3736ec888 ("cpu/hotplug:
Hide cpu_up/down()"), causing build failure.

Use the safer add_cpu() and remove_cpu() instead.

Fixes: "AOSCOS: MIPS: Loongson 3: Introduce disable_unused_cpus() to save power"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/loongson64/smp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/mips/loongson64/smp.c b/arch/mips/loongson64/smp.c
index f0847ef7fc8d..3690e4bb0ad5 100644
--- a/arch/mips/loongson64/smp.c
+++ b/arch/mips/loongson64/smp.c
@@ -850,10 +850,10 @@ int disable_unused_cpus(void)
 	cpumask_and(&tmp, &tmp, cpu_possible_mask);
 
 	for_each_cpu(cpu, &tmp)
-		cpu_up(cpu);
+		add_cpu(cpu);
 
 	for_each_cpu(cpu, &tmp)
-		cpu_down(cpu);
+		remove_cpu(cpu);
 
 	return 0;
 }
-- 
2.47.1

